### PR TITLE
NOISSUE - Refactor TLS layer

### PR DESCRIPTION
**Pull request by dborovcanin**
Pull request title should be `MF-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](https://github.com/absmach/magistrala/blob/master/CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
This PR updates TLS layer in mProxy.

### Which issue(s) does this PR fix/relate to?
There is no such issue.

### List any changes that modify/break current functionality
TLS and CRL/OCSP layers are improved.

### Have you included tests for your changes?
No.

### Did you document any new/modified functionality?
N/A

### Notes
N/A

### CODE REVIEW

File: pkg/tls/verifications.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func NewVerification(opts env.Options) ([]verifier.Verifier, error) {
+	if opts.FuncMap == nil {
+		opts.FuncMap = make(map[reflect.Type]env.ParserFunc)
+	}
+	opts.FuncMap[reflect.TypeOf(make([]verification, 0))] = envParseSliceValidate
+	opts.FuncMap[reflect.TypeOf(new(verification))] = envParseValidation
+
+	var c struct {
+		Verifications []verification `env:"_CERT_VERIFICATION_METHODS"             envDefault:""`
+	}
```
Comment (arvindh123) [inline pkg/tls/verifications.go:37]: mTLS verification method is not loading properly from .env 

```suggestion
		Verifications []verification `env:"CLIENT_CERT_VERIFICATION_METHODS"             envDefault:""`
```

and also 
Please update .env to 
`MPROXY_MQTT_WITH_MTLS_CLIENT_CERT_VERIFICATION_METHODS=ocsp`
`MPROXY_MQTT_WS_WITH_MTLS_CLIENT_CERT_VERIFICATION_METHODS=ocsp`
`MPROXY_HTTP_WITH_MTLS_CLIENT_CERT_VERIFICATION_METHODS=ocsp`

File: config.go [hunk 2]
```diff
@@ -16,24 +15,25 @@ type Config struct {
...
 func NewConfig(opts env.Options) (Config, error) {
 	c := Config{}
 	if err := env.ParseWithOptions(&c, opts); err != nil {
 		return Config{}, err
 	}
-	mptlsConfig, err := mptls.NewConfig(opts)
+	vfs, err := mptls.NewVerification(opts)
+	if err != nil {
+		return Config{}, err
+	}
```
Comment (arvindh123) [inline config.go]: @dborovcanin 
This is part of TLS config and Why are keeping out of mptls.NewConfig ?

**Review by arvindh123 — COMMENTED** [review]
@dborovcanin

Code now looks clean and readable.

**Review by arvindh123 — COMMENTED** [review]
@dborovcanin

Code now looks clean and readable.

**Review by arvindh123 — COMMENTED** [review]
@dborovcanin

Code now looks clean and readable.

File: config.go (hunk 2 above)
Comment (dborovcanin) [reply config.go]: Fixed, please check now.

File: examples/client/http/with_mtls.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 protocol=https
 host=localhost
-port=8088
+port=8008
 path="messages"
```
Comment (arvindh123) [inline examples/client/http/with_mtls.sh]: Please change value to `8088`, because http mtls server in `cmd/main.go` run at 8088

**Review by arvindh123 — APPROVED** [review]
LGTM

**Pull request merged**